#!/usr/bin/env python
PACKAGE = "graspberry_robot"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("profile_vel_0",	double_t,	0,	"Profile velocity for joint 0",	10.0, 0.0, 0xFFFFFFFF)
gen.add("profile_vel_1",	double_t,	0,	"Profile velocity for joint 1",	20.0, 0.0, 0xFFFFFFFF)
gen.add("profile_vel_2",	double_t,	0,	"Profile velocity for joint 2",	20.0, 0.0, 0xFFFFFFFF)

gen.add("profile_acc_0",	double_t,	0,	"Profile acceleration for joint 0",	10.0, 0.0, 0xFFFF)
gen.add("profile_acc_1",	double_t,	0,	"Profile acceleration for joint 1",	10.0, 0.0, 0xFFFF)
gen.add("profile_acc_2",	double_t,	0,	"Profile acceleration for joint 2",	10.0, 0.0, 0xFFFF)

gen.add("profile_dec_0",	double_t,	0,	"Profile deceleration for joint 0",	10.0, 0.0, 0xFFFF)
gen.add("profile_dec_1",	double_t,	0,	"Profile deceleration for joint 1",	10.0, 0.0, 0xFFFF)
gen.add("profile_dec_2",	double_t,	0,	"Profile deceleration for joint 2",	10.0, 0.0, 0xFFFF)

gen.add("homing",			bool_t,		0,	"Home the arm at the startup",	False)

gen.add("homing_vel_0",	double_t,	0,	"Homing velocity for joint 0",	5.0, 0.0, 0xFFFFFFFF)
gen.add("homing_vel_1",	double_t,	0,	"Homing velocity for joint 1",	20.0, 0.0, 0xFFFFFFFF)
gen.add("homing_vel_2",	double_t,	0,	"Homing velocity for joint 2",	5.0, 0.0, 0xFFFFFFFF)

exit(gen.generate(PACKAGE, "graspberry_robot", "GraspberryArm"))
